### PR TITLE
fix(android)(8_1_X): set ListItem properties into proxy properties

### PR DESCRIPTION
**Pull request by garymathews**
- Cherry-pick of https://github.com/appcelerator/titanium_mobile/pull/10978

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-24415)

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIDialog.java [hunk 1]
```diff
@@ -285,7 +285,7 @@ public void propertyChanged(String key, Object oldValue, Object newValue, KrollP
 						}
 						ViewCompat.setImportantForAccessibility(listView, importance);
 					} else {
-						listView.setContentDescription(composeContentDescription());
+						listView.setContentDescription(getProxy().composeContentDescription());
```
Comment (sgtcoolguy) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIDialog.java]: can getProxy() return null?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListSectionProxy.java [hunk 2]
```diff
@@ -675,12 +678,18 @@ public void populateViews(KrollDict data, TiBaseListViewItem cellContent, TiList
 				KrollDict properties = new KrollDict((HashMap) data.get(binding));
 				KrollDict diffProperties = viewItem.generateDiffProperties(properties);
 				if (!diffProperties.isEmpty()) {
+					if (viewProxy != null && viewProxy.getProperties() != null) {
```
Comment (sgtcoolguy) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListSectionProxy.java]: extract temp var to assign viewProxy.getProperties() value to?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListSectionProxy.java [hunk 3]
```diff
@@ -675,12 +678,18 @@ public void populateViews(KrollDict data, TiBaseListViewItem cellContent, TiList
...
+						viewProxy.getProperties().putAll(diffProperties);
+					}
 					view.processProperties(diffProperties);
 				}
 
 			} else if (dataItem != null && view != null) {
 				KrollDict diffProperties = viewItem.generateDiffProperties(dataItem.getDefaultProperties());
 				if (!diffProperties.isEmpty()) {
+					if (viewProxy != null && viewProxy.getProperties() != null) {
```
Comment (sgtcoolguy) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListSectionProxy.java]: same as above

File: android/titanium/src/java/org/appcelerator/kroll/KrollProxy.java [hunk 4]
```diff
@@ -978,6 +976,57 @@ public boolean doFireEvent(String event, Object data)
...
+	 * We combine these to create the single Android property contentDescription.
+	 * (e.g., View.setContentDescription(...));
+	 */
+	public String composeContentDescription()
+	{
+		if (properties == null) {
+			return null;
+		}
+
+		final String punctuationPattern = "^.*\\p{Punct}\\s*$";
```
Comment (sgtcoolguy) [inline android/titanium/src/java/org/appcelerator/kroll/KrollProxy.java]: Extract to a constant so it's compiled once?

File: android/titanium/src/java/org/appcelerator/kroll/KrollProxy.java [hunk 5]
```diff
@@ -978,6 +976,57 @@ public boolean doFireEvent(String event, Object data)
...
+		StringBuilder buffer = new StringBuilder();
+		String label = TiConvert.toString(properties.get(TiC.PROPERTY_ACCESSIBILITY_LABEL));
+		String hint = TiConvert.toString(properties.get(TiC.PROPERTY_ACCESSIBILITY_HINT));
+		String value = TiConvert.toString(properties.get(TiC.PROPERTY_ACCESSIBILITY_VALUE));
+
+		if (!TextUtils.isEmpty(label)) {
```
Comment (sgtcoolguy) [inline android/titanium/src/java/org/appcelerator/kroll/KrollProxy.java]: These three small blocks are identical, extract to a method? Only wrinkle is the extra space prefix/suffix added if previous value was not empty...
```
buffer.append(composeContentSegment(TiC.PROPERTY_ACCESSIBILITY_LABEL));
buffer.append(composeContentSegment(TiC.PROPERTY_ACCESSIBILITY_HINT));
buffer.append(composeContentSegment(TiC.PROPERTY_ACCESSIBILITY_VALUE));
```

**Review by sgtcoolguy — APPROVED** [review]
LGTM, but could use some very minor refactors/cleanup.

**Comment by lokeshchdhry** [comment]
FR Passed.

Accessibility label in the ListItem components. Enabling text to speech speaks out the accessibility label & value as expected.

Studio Ver: 5.1.3.201906102126
SDK Ver: 8.1.0 local build
OS Ver: 10.14.5
Xcode Ver: Xcode 10.2.1
Appc NPM: 4.2.14-3
Appc CLI: 7.1.0-master.24
Daemon Ver: 1.1.3
Ti CLI Ver: 5.2.1
Alloy Ver: 1.14.0
Node Ver: 8.15.1
NPM Ver: 6.4.1
Java Ver: 10.0.2
Devices: ⇨ google Pixel (Android 9)

**Pull request merged**